### PR TITLE
Adding support for ingress

### PR DESCRIPTION
**Pull request by ageekymonk**
Added support for ingress

### CODE REVIEW

**Comment by chrisbarrett** [comment]
Thanks very much for the PRs! ❤️ 

I want to look at the failing CI before I do a review and pull in your changes. Once that's done I'll review this and your other PR. :)

**Comment by ageekymonk** [comment]
@chrisbarrett Do you want me to do anything else

File: kubernetes-ingress.el [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+(require 'kubernetes-props)
+(require 'kubernetes-state)
+(require 'kubernetes-utils)
+(require 'kubernetes-yaml)
+
+
+;; Components
+
+(defconst kubernetes-ingress--column-heading
+  (propertize (format "%-45s %6s" "Name" "Age")
```
Comment (noorul) [inline kubernetes-ingress.el]: Can we add hosts, address and ports also?

File: kubernetes-ingress.el [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  `(section (,(intern (kubernetes-state-resource-name ingress)) t)
+            (heading (ingress-line ,state ,ingress))
+            (section (details nil)
+                     (indent
+                      (ingress-detail ,ingress)
+                      (padding)))))
+(kubernetes-ast-define-component ingress-list (state &optional hidden)
+  (-let [(&alist 'items ingress) (kubernetes-state-ingress state)]
+    `(section (ingress-container ,hidden)
+              (header-with-count "ingress" ,ingress)
```
Comment (noorul) [inline kubernetes-ingress.el]: nit, All the header starts with  capital letters. I think if we use "Ingress" we can be consistent.

File: kubernetes-kubectl.el [hunk 3]
```diff
@@ -276,6 +293,23 @@ ERROR-CB is called if an error occurred."
...
+(defun kubernetes-kubectl-delete-ingress (props state ingress-name cb &optional error-cb)
+  "Delete INGRESS-NAME, then execute CB with the response buffer.
+
+PROPS is an alist of functions to inject.  It should normally be passed
+`kubernetes-props'.
+
+STATE is the application state.
+
+ERROR-CB is called if an error occurred."
+  (kubernetes-kubectl props state (list "delete" "ingress" secret-name "-o" "name")
```
Comment (noorul) [inline kubernetes-kubectl.el]: I think copy paste typo here. This should be ingress-name.

**Comment by noorul** [comment]
@ageekymonk I see that there are tests under test folder for other k8s resources. Can we add one for ingress?

File: kubernetes-ingress.el (hunk 1 above)
Comment (ageekymonk) [reply kubernetes-ingress.el]: Added Hosts and Address. Ports was always 80. So ignored it.

File: kubernetes-ingress.el (hunk 2 above)
Comment (ageekymonk) [reply kubernetes-ingress.el]: Thanks. Updated it.

File: kubernetes-kubectl.el (hunk 3 above)
Comment (ageekymonk) [reply kubernetes-kubectl.el]: Thanks Updated it

**Comment by ageekymonk** [comment]
Thanks @noorul . I have updated the tests and added what you have requested. Let me know if you are okay with it.

**Comment by noorul** [comment]
@ageekymonk The buffer is not formatted properly

File: kubernetes-ingress.el [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+          (line `(line ,(concat
+                         ;; Name
+                         (format "%-45s " (kubernetes-utils-ellipsize name 45))
+
+                         ;; Hosts
+                         (format "%-25s " (--mapcat (alist-get 'host it) ingress-rules))
+
+                         ;; Address
+                         (format "%20s " (--mapcat (alist-get 'ip it) ingress-lb-list))
+
```
Comment (noorul) [inline kubernetes-ingress.el]: Ports is missing.

**Comment by noorul** [comment]
See the buffer that I am getting


kubernetes-el-ingress

File: kubernetes-ingress.el (hunk 4 above)
Comment (ageekymonk) [reply kubernetes-ingress.el]: Can you attach the ingress json ? Will work with that.

Comment (noorul) [reply kubernetes-ingress.el]: @ageekymonk I am afraid I cant because it has company information.

Comment (noorul) [reply kubernetes-ingress.el]: Or may be I can mask it with random strings. Let me try.

Comment (noorul) [reply kubernetes-ingress.el]: @ageekymonk I am finding it difficult to remove all the important information from the yaml. Maybe what you need is a longer host name

**Comment by noorul** [comment]
Closing this as ingress support is now in master.

**Pull request closed without merging**